### PR TITLE
add Answer page

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add code for the design of the Answers page and also the 404 page

In this PR, **421 additions 169 deletions** lines of code
- some files contain only 5 to 6 lines of code
- some files are of Images, so you can avoid them

### Please review this PR and give your feedback

---

#### [Live URL](https://react-marvels-quiz-git-page-answer-hsnice16.vercel.app/quiz/answers?category=avengers-infinity-war)

image
image

### CODE REVIEW

File: src/components/button/Button.jsx [hunk 1]
```diff
@@ -2,29 +2,42 @@ import "./button.css";
...
-      className={classNames(
-        `${iconDirection}-double-angle-icon`,
-        "fas",
-        `fa-angle-double-${iconDirection}`
-      )}
-    ></i>
+    {btnText === "Back to home" ? (
+      <>
+        <i
+          className={classNames(
```
Comment (Developers-codz) [inline src/components/button/Button.jsx:10]: Appreciation: use of classnames

File: src/pages/question/ShowQuestion.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                : `uestion ${capitalizeFirstLetter(prev)}`}
+            </span>
+          </Link>
+
+          <Link to={nextLink} className="fw-bold link ml-auto pb-1">
+            <KushanFontFamily textToShow={next === "result" ? "R" : "Q"} />
+            <span className="text-slate-800">
+              {next === "result"
+                ? "esult"
+                : `uestion ${capitalizeFirstLetter(next)}`}{" "}
```
Comment (Developers-codz) [inline src/pages/question/ShowQuestion.jsx:86]: don't understood this part, What exactly this code is doing here

**Review by Developers-codz — APPROVED** [review]
Everything looks good to go 
Great work on code

File: src/components/button/Button.jsx [hunk 3]
```diff
@@ -2,29 +2,42 @@ import "./button.css";
...
+            "fas",
+            "fa-angle-double-right"
+          )}
+        ></i>
+      </>
+    )}
   </button>
 );
 
 Button.propTypes = {
```
Comment (bhtibrewal) [inline src/components/button/Button.jsx:33]: Appreciation: good work defining the propTypes

File: src/pages/answers/Answers.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  useDocumentTitle(`${name}, Answers`);
+  useScrollToTop();
+
+  return (
+    <>
+      <div className="fs-lg mb-4 mt-2 text-center">{name}</div>
+
+      <h1 className="ctr-text">
+        <span className="py-2">
```
Comment (bhtibrewal) [inline src/pages/answers/Answers.jsx:24]: Why have you used this extra span tag?

File: src/pages/answers/Answers.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                category={category}
+                onPage="Answers"
+              />
+            </section>
+          );
+        })}
+
+        <Button
+          onClick={() => {
+            navigate(ROUTE_ROOT);
```
Comment (bhtibrewal) [inline src/pages/answers/Answers.jsx:61]: Appreciate the use of var here

**Review by bhtibrewal — APPROVED** [review]
Well written code, excellent use of coding conventions, you are good to merge

File: src/pages/question/ShowQuestion.jsx (hunk 2 above)
Comment (hsnice16) [reply src/pages/question/ShowQuestion.jsx:86]: yeah!
As on the first question, I want to show links for `rules` and `question two`, And on the last question I want to show links for `question four` and `result`. So this code is handling that thing.

If we are not on the first question or last question, then show links for the `next question` and `previous question`.

File: src/pages/answers/Answers.jsx (hunk 4 above)
Comment (hsnice16) [reply src/pages/answers/Answers.jsx:24]: Yeah, 
As I am designing the underline before and after the text using the `::before` and `::after` na, so I want some space between text and underline that's why I enclose the whole text inside a span and then gave padding in the span. (pic)
image

And another reason is I'm using `flex` to align the `::before`, text, and `::after` element and the `flex-direction` is `column`. So, I will not enclose the text inside a `span`, the single character `A` will come on one line and rest of the characters will come on a separate line. As I'm using `span` to design `A` differently.

**Comment by hsnice16** [comment]
Thank you for reviewing,

**Pull request merged**